packages/Interface/src/DTK_C_API.cpp

```diff
@@ -0,0 +1,434 @@
+                               unsigned &field_dimension,
+                               size_t &local_num_dofs )
+{
+    auto u = get_function<DTK_field_size_function>( user_data );
+    u.first( u.second, field_name.c_str(), &field_dimension, &local_num_dofs );
+}
+
+template <class Scalar>
+void PullFieldDataFunctionWrapper( std::shared_ptr<void> user_data,
+                                   const std::string &field_name,
+                                   View<Scalar> field_dofs )
+{
+    throw DataTransferKitException( "Not implemented" );
+}
+template <>
+void PullFieldDataFunctionWrapper<double>( std::shared_ptr<void> user_data,
+                                           const std::string &field_name,
+                                           View<double> field_dofs )
+{
+    auto u = get_function<DTK_pull_field_data_function>( user_data );
+    u.first( u.second, field_name.c_str(), field_dofs.data() );
+}
+
+template <class Scalar>
+void PushFieldDataFunctionWrapper( std::shared_ptr<void> user_data,
+                                   const std::string &field_name,
+                                   const View<Scalar> field_dofs )
+{
+    throw DataTransferKitException( "Not implemented" );
+}
+template <>
+void PushFieldDataFunctionWrapper<double>( std::shared_ptr<void> user_data,
+                                           const std::string &field_name,
+                                           const View<double> field_dofs )
+{
+    auto u = get_function<DTK_push_field_data_function>( user_data );
+    u.first( u.second, field_name.c_str(), field_dofs.data() );
+}
+
+template <class Scalar>
+void EvaluateFieldFunctionWrapper( std::shared_ptr<void> user_data,
+                                   const std::string &field_name,
+                                   const View<Coordinate> evaluation_points,
+                                   const View<LocalOrdinal> object_ids,
+                                   View<Scalar> values )
+{
+    throw DataTransferKitException( "Not implemented" );
+}
+template <>
+void EvaluateFieldFunctionWrapper<double>(
+    std::shared_ptr<void> user_data, const std::string &field_name,
+    const View<Coordinate> evaluation_points,
+    const View<LocalOrdinal> object_ids, View<double> values )
+{
+    auto u = get_function<DTK_evaluate_field_function>( user_data );
+    u.first( u.second, field_name.c_str(), evaluation_points.data(),
+             object_ids.data(), values.data() );
+}
+
+} // namespace DataTransferKit
+
+extern "C" {
+
+const char *DTK_version()
+{
+    static std::string version_string = DataTransferKit::version().c_str();
+    return version_string.c_str();
+}
+
+const char *DTK_git_commit_hash()
+{
+    static std::string hash_string = DataTransferKit::git_commit_hash().c_str();
+    return hash_string.c_str();
+}
+
+DTK_Handle DTK_create( DTK_EXECUTION_SPACE space )
+{
+    return new DataTransferKit::DTK_Struct( space );
```

Comment:
The problem is with the `new` and `delete` it is easy to leak. If I do: 
```C
DTK_Handle h = DTK_create(a);
h = DTK_create(b);
DTK_destroy(&h);
```
you will start leaking and if I do:
```C
DTK_Handle h;
DTK_destroy(&h);
```
depending on what the compiler does with `h`, it may be fine or it will try to free memory it doesn't own.